pkg/process/checks/process_fallback.go

```diff
@@ -0,0 +1,21 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !linux
+
+package checks
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// processesByPID returns the processes by pid from the system probe
```

Comment:
Its not from the system probe, process probe is just the name of the interface. The system probe is only used to collect IO stats, if enabled. Confusing name, I suppose.
```suggestion
// processesByPID returns the processes by pid from the process probe
```

Reply:
Also, some more context would be good on why we need different versions per build.

Reply:
That is odd that the probe has the ability to to try to get those stats since we call for it from the system probe [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/process/checks/process.go#L258) which collects it [here](https://github.com/DataDog/datadog-agent/blob/main/cmd/system-probe/modules/process.go#L80). Not sure in which case `p.elevatedPermissions` is true and the stats can actually be collected.

Reply:
`p.elevatedPermissions` is true when the process module is enabled for the system probe: 

https://github.com/DataDog/datadog-agent/blob/a747ad4ce42c8087210a5998379f76e26da5a2b2/pkg/process/checks/process.go#L149-L150

so it appears that we are getting privileged data twice. Made a loose ends ticket to fix this bug: https://datadoghq.atlassian.net/browse/CXP-2488

Reply:
I changed the title, but unfortunately, the process probe does work after testing locally and enabling the system probe.

Reply:
ah yes that makes sense